catalog/imaging/imaging-transformer-chipping/src/main/java/org/codice/alliance/imaging/chip/transformer/CatalogOutputAdapter.java

```diff
@@ -239,26 +277,33 @@ private void copySDEs(TaggedRecordExtensionHandler in, TaggedRecordExtensionHand
   public BinaryContent getNitfBinaryContent(
       BufferedImage chip, NitfSegmentsFlow nitfSegmentsFlow, int sourceX, int sourceY)
       throws IOException, MimeTypeParseException, NitfFormatException {
-
     try {
-      NitfHeader chipHeader = createChipHeader(nitfSegmentsFlow);
+      lock.acquire();
+      try {
+        NitfHeader chipHeader = createChipHeader(nitfSegmentsFlow);
 
-      nitfSegmentsFlow.fileHeader(nitfHeader -> copySDEs(nitfHeader, chipHeader));
+        nitfSegmentsFlow.fileHeader(nitfHeader -> copySDEs(nitfHeader, chipHeader));
 
-      ImageSegment chipImageSegment =
-          createChipImageSegment(chip, sourceX, sourceY, nitfSegmentsFlow);
+        ImageSegment chipImageSegment =
+            createChipImageSegment(chip, sourceX, sourceY, nitfSegmentsFlow);
 
-      addIchipbTre(chip, sourceX, sourceY, chipImageSegment);
+        addIchipbTre(chip, sourceX, sourceY, chipImageSegment);
 
-      List<ImageSegment> imageSegments = getImageSegments(nitfSegmentsFlow);
-      if (!imageSegments.isEmpty()) {
-        copySDEs(imageSegments.get(0), chipImageSegment);
-      }
+        List<ImageSegment> imageSegments = getImageSegments(nitfSegmentsFlow);
+        if (!imageSegments.isEmpty()) {
+          copySDEs(imageSegments.get(0), chipImageSegment);
+        }
 
-      return nitfToBinaryContent(chipHeader, chipImageSegment);
-    } finally {
-      nitfSegmentsFlow.end();
+        return nitfToBinaryContent(chipHeader, chipImageSegment);
+      } finally {
+        nitfSegmentsFlow.end();
```

Comment:
❓ Can this throw an exception? If so, the lock won't be released.

Reply:
I didn't think this could, but after digging down into it, it looks like it might actually be possible, since that `end()` method is actually just calling `run()` on a `Runnable` which could throw anything.